andela/knights-bn-backend
# 170947680 - Send Notification for Posted Comments

#### What does this PR do?

- Notify a user when a new comment is made on the request.

#### Description of Task to be completed?

- A manager should post a comment on a request
- A requester should see a notification on a posted comment

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git checkout ft-notifications-for-comments-170947680 ` to switch to this branch.
- run `npm install` to install dependencies.
- go to this link to see live notification `https://codepen.io/moise1/pen/jOPYveW`
- create a .env file then add environment variable as shown in ` .env.example`,  check our slack channel `code-knights-12` for .env

```
   MANAGER_PSWD=$2b$10$SO7hwwylEnHKZIv68w/ePOK4yybPnhXa5.tQSUBe.PSb6FBKyuVO.
```

- create two Postgres database for development and then test
- run `npm run createTables` to create tables for you

- run `npm run dev-start` to start the server.

- login using existing user credentials POST `localhost:4000/api/v1/auth/login`

```
{
    "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}

```
- copy the `user's token` in the following app `https://codepen.io/moise1/pen/jOPYveW`

- make a new request by POST `http://localhost:4000/api/v1/trips/returnTrip`


{
	"origin": "Burundo",
	"destination": "kgl",
	"departureDate": "9002-04-02",
	"returnDate": "9003-05-18",
	"accommodation": "XYZ campus",
	"reason": "Having a meeting",
	"passportNumber": "yuyut56"
}
### - log in as a manager using these credentials:

```
{
    "email": "william.ishimwe@andela.com",
    "password": "Password@1"
    }
```

### Use the following in your Postman: 

### Request ID to comment on in the Params tab
<img width="1005" alt="Screen Shot 2020-03-10 at 18 01 25" src="https://user-images.githubusercontent.com/25052255/76332624-45769700-62f9-11ea-8e0d-bf562008c90a.png">

#### Comment in the body tab
<img width="1148" alt="Screen Shot 2020-03-10 at 17 59 28" src="https://user-images.githubusercontent.com/25052255/76332399-ff213800-62f8-11ea-8d24-1390356e4486.png">

##### visit this codepen again to see the result: [CodePen](https://codepen.io/moise1/pen/jOPYveW)
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

[#170947680](https://www.pivotaltracker.com/story/show/170947680)
